kindtest.sh

```diff
@@ -0,0 +1,187 @@
+# around this limitation by running a Docker registry in a Docker container and connecting the networks so
+# that the Kubernetes nodes have visibility to the registry. When you run tests, you will see that Docker
+# images are pushed to this local registry and then are pulled to the Kubernetes nodes as needed.
+# You can see the images on a node by running: "docker exec -it kind-worker crictl images" where kind-worker
+# is the name of the Docker container.
+#
+# As of May 6, 2020, the tests are clean on Kubernetes 1.16 with the following JDK workarounds:
+# 1. Maven must be run with OpenJDK 11.0.7, available here: https://github.com/AdoptOpenJDK/openjdk11-upstream-binaries/releases/download/jdk-11.0.7%2B10/OpenJDK11U-jdk_x64_linux_11.0.7_10.tar.gz
+#    This is because of a critical bug fix. Unfortunately, the Oracle JDK 11.0.7 release was based on an earlier build and doesn't have the fix.
+# 2. The WebLogic Image Tool will not accept an OpenJDK JDK. Set WIT_JAVA_HOME to an Oracle JDK Java Home.
+#    For example, "export WIT_JAVA_HOME=/usr/java/jdk-11.0.7" before running this script.
+#
+# If you want to access the cluster, use "kubectl cluster-info --context kind-kind" where the trailing "kind" is the value of the "-n" argument.
+set -o errexit
+
+script="${BASH_SOURCE[0]}"
+scriptDir="$( cd "$( dirname "${script}" )" && pwd )"
+
+function usage {
+  echo "usage: ${script} [-v <version>] [-n <name>] [-o <directory>] [-t <tests>] [-h]"
+  echo "  -v Kubernetes version (optional) "
+  echo "      (default: 1.15.11, supported values: 1.18, 1.18.2, 1.17, 1.17.5, 1.16, 1.16.9, 1.15, 1.15.11, 1.14, 1.14.10) "
+  echo "  -n Kind cluster name (optional) "
+  echo "      (default: kind) "
+  echo "  -o Output directory (optional) "
+  echo "      (default: \${WORKSPACE}/logdir/\${BUILD_TAG}, if \${WORKSPACE} defined, else /scratch/\${USER}/kindtest) "
+  echo "  -t Test filter (optional) "
+  echo "      (default: **/It*) "
+  echo "  -h Help"
+  exit $1
+}
+
+k8s_version="1.15.11"
+kind_name="kind"
+if [[ -z "${WORKSPACE}" ]]; then
+  outdir="/scratch/${USER}/kindtest"
+else
+  outdir="${WORKSPACE}/logdir/${BUILD_TAG}"
+fi
+test_filter="**/It*"
+
+while getopts ":h:n:o:t:v:" opt; do
+  case $opt in
+    v) k8s_version="${OPTARG}"
+    ;;
+    n) kind_name="${OPTARG}"
+    ;;
+    o) outdir="${OPTARG}"
+    ;;
+    t) test_filter="${OPTARG}"
+    ;;
+    h) usage 0
+    ;;
+    *) usage 1
+    ;;
+  esac
+done
+
+function versionprop {
+  grep "${1}=" "${scriptDir}/kindversions.properties"|cut -d'=' -f2
+}
+
+kind_image=$(versionprop "${k8s_version}")
+if [ -z "${kind_image}" ]; then
+  echo "Unsupported Kubernetes version: ${k8s_version}"
+  exit 1
+fi
+
+echo "Using Kubernetes version: ${k8s_version}"
+
+mkdir -m777 -p "${outdir}"
+export RESULT_ROOT="${outdir}/wl_k8s_test_results"
+if [ -d "${RESULT_ROOT}" ]; then
+  rm -Rf "${RESULT_ROOT}/*"
```

Comment:
RESULT_ROOT is set just above,  but never-the-less an rm -rf $ENV/* of env var looks a bit like playing with fire to me.  A "set -u" would make it a little safer maybe.  Putting the 'wl_k8s_test_results' string directly  in the rm expression makes it even safer IMO.  Ditto for PV_ROOT below.  This reduces chance that a future change  to the env var logic could leave a crater in the file system.